library/seq.lisp

```diff
@@ -207,6 +209,30 @@ a new `Seq` instance."
                    (rebalance-branches
                     (fold <> (butlast lsubts) (make-list nsubts (butfirst rsubts))))))))))))
 
+  (declare split ((Eq :a) (types:RuntimeRepr :a) => :a -> (Seq :a) -> (iter:Iterator (Seq :a))))
```

Comment:
This implementation needs to be streaming. The current version won't work with infinite iterators.

Reply:
I'm confused about this- 

I don't believe`Seq` can be infinite, so I'm not sure how a `seq` could or should generate an infinite/stream `Iterator`.

Reply:
Or do you mean simply that I should also define split for `Iterator` and ensure infinite compatibility?